php_phongo.c

```diff
@@ -590,11 +590,15 @@ int phongo_execute_query(mongoc_client_t *client, const char *namespace, zval *z
 	return true;
 } /* }}} */
 
-static bson_t *create_wrapped_command_envelope(bson_t *reply)
+static bson_t *create_wrapped_command_envelope(const char *db, bson_t *reply)
 {
 	bson_t *tmp;
+	size_t max_ns_len = strlen(db) + 5 + 1; /* db + ".$cmd" + '\0' */
+	char *ns = emalloc(max_ns_len);
 
-	tmp = BCON_NEW("cursor", "{", "id", BCON_INT64(0), "ns", BCON_UTF8("db.$cmd"), "firstBatch", "[", BCON_DOCUMENT(reply), "]", "}");
+	snprintf(ns, max_ns_len, "%s.$cmd", db);
```

Comment:
Random thought: if `db` contains null bytes, I assume `snprintf()` would simply stop appending at the character immediately preceding the first null byte. In this case, I suppose the only side effect is that we allocated more memory for `ns` than was needed. More importantly, this won't affect `BCON_NEW()` below.

libmongoc also doesn't validate the database name when executing a command (and it also appears to assume it's null-terminated when building the command parts), so I expect it'd be up to the server to report a possible error (really only if the command required a database to exist and it did not).

Reply:
It doesn't matter in any case, as the value will never be used for a getMore that might need it, as the cursorID is ``0``. Allocating a few more bytes won't matter either, and is certainly going to be faster than checking for NULL bytes. We don't have access to the length in phongo_execute_command either although we could of course add that (for no, gain).